### PR TITLE
Post commit review feedback from 4ba6a505639a5d9bcdd7bc1a4bdfb0c8131d…

### PR DESCRIPTION
**Pull request by gottesmm**
…6a67 (#18017).

rdar://42301529

### CODE REVIEW

**Comment by gottesmm** [comment]
Just an FYI, I needed to change the internal set representation to be a StringSet since otherwise I won't be able to print out the names of deleted functions that the analysis was not notified of. Shouldn't make a difference in terms of performance.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: lib/SILOptimizer/Analysis/PassManagerVerifierAnalysis.cpp [hunk 1]
```diff
@@ -61,33 +56,48 @@ void PassManagerVerifierAnalysis::notifyWillDeleteFunction(SILFunction *f) {
...
+  // going through the module's function list and attempting to remove all
+  // functions in the module. If we fail to remove fn, then we know that a
+  // function was added to the module without an appropriate message being sent
+  // by the pass manager.
+  bool foundError = false;
   for (auto &fn : mod) {
-    ++funcCount;
-    assert(liveFunctions.count(&fn) &&
-           "Found function in module that verifier is not aware of?!");
+    if (liveFunctions.count(fn.getName())) {
```
Comment (atrick) [inline lib/SILOptimizer/Analysis/PassManagerVerifierAnalysis.cpp]: Did you mean to call `erase` here?

Comment (gottesmm) [reply lib/SILOptimizer/Analysis/PassManagerVerifierAnalysis.cpp]: No.

**Comment by gottesmm** [comment]
For those listening in, I found a much simpler way to do this. I am going to add a function on the passmanager for adding/removing functions. That will enable the pass manager to send the correct messages and not require passes to notify it about creating/removing functions. Instead the passes only need to notify the pass manager if it wants the pass to be "rerun" (in the sense of being a function pass).

This is much easier to not mess up.

**Pull request closed without merging**